File: core/src/main/java/io/github/fourlastor/game/demo/round/step/SimpleStep.java
package io.github.fourlastor.game.demo.round.step;

import io.github.fourlastor.game.demo.state.GameState;
import java.util.function.Consumer;

public abstract class SimpleStep extends Step<Void> {

    public abstract void enter(GameState state, Runnable continuation);

    @Override
    public void enter(GameState state, Consumer<Void> continuation) {
        enter(state, () -> continuation.accept(null));
    }
}

File: core/src/main/java/io/github/fourlastor/game/demo/round/step/SimpleStep.kt
package io.github.fourlastor.game.demo.round.step

import io.github.fourlastor.game.demo.state.GameState
import java.util.function.Consumer

abstract class SimpleStep : Step<Unit>() {
    abstract fun enter(state: GameState, continuation: Runnable)
    override fun enter(state: GameState, continuation: Consumer<Unit>, cancel: Runnable) {
        enter(state) { continuation.accept(Unit) }
    }
}
